pkg/client/client.go

```diff
@@ -41,21 +50,22 @@ func NewTakClient(ctx context.Context, config ClientConfig) (*TakClient, error)
 		return nil, fmt.Errorf("port must be between 1 and 65535")
 	}
 
-	// Connect to the server
-	conn, err := net.Dial("tcp", net.JoinHostPort(config.Host, fmt.Sprint(config.Port)))
+	addr := net.JoinHostPort(config.Host, fmt.Sprint(config.Port))
+	conn, err := dialWithBackoff(ctx, addr, config.Logger, config.Reconnect)
 	if err != nil {
-		return nil, fmt.Errorf("failed to connect to server: %v", err)
+		return nil, fmt.Errorf("failed to connect: %w", err)
 	}
 
 	client := TakClient{
 		Conn:     conn,
 		sendChan: make(chan []byte, 50),
 	}
 
-	if config.Logger == nil {
-		client.Logger = zerolog.New(zerolog.ConsoleWriter{Out: os.Stdout}).With().Timestamp().Logger()
+	if reflect.ValueOf(config.Logger).IsZero() {
```

Comment:
I generally try to avoid reflection but I wanted to make the use of Logger consistent, it was a pointer in one struct and a value in another so I decided to make them both values since that's generally more performant overall. Took the trade on zero struct compare w/ reflection since it's not *that* expensive and its only on connect